pocs/camera/camera.py

```diff
@@ -155,14 +172,99 @@ def CCD_cooling_power(self):
 # Methods
 ##################################################################################################
 
-    def take_observation(self, *args, **kwargs):
-        raise NotImplementedError
+    def take_observation(self, observation, headers=None, filename=None, *args, **kwargs):
+        """Take an observation
+
+        Gathers various header information, sets the file path, and calls
+            `take_exposure`. Also creates a `threading.Event` object and a
+            `threading.Thread` object. The Thread calls `process_exposure`
+            after the exposure had completed and the Event is set once
+            `process_exposure` finishes.
+
+        Args:
+            observation (~pocs.scheduler.observation.Observation): Object
+                describing the observation
+            headers (dict): Header data to be saved along with the file.
+            **kwargs (dict): Optional keyword arguments (`exp_time`, dark)
+
+        Returns:
+            threading.Event: An event to be set when the image is done processing
+        """
+        # To be used for marking when exposure is complete (see `process_exposure`)
+        camera_event = Event()
+
+        exp_time, file_path, image_id, metadata = self._setup_observation(observation,
+                                                                          headers,
+                                                                          filename,
+                                                                          *args,
+                                                                          **kwargs)
+
+        exposure_event = self.take_exposure(seconds=exp_time, filename=file_path, *args, **kwargs)
+
+        # Add most recent exposure to list
+        if self.is_primary:
+            observation.exposure_list[image_id] = file_path
+
+        # Process the exposure once readout is complete
+        t = Thread(target=self.process_exposure, args=(metadata, camera_event, exposure_event))
```

Comment:
I've been wondering lately if we should return this thread process along with the `Event`. Might be nice to be able to hold on to it in case we need to join or kill or something. Any experience with needing that on the SBIGs or FLIs?

Reply:
I don't recall any issues with the `process_exposure` thread getting stuck. The readout threads are a different matter, though, seen those get stuck many times due to USB problems.